servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/server/Karaf4xServerDelegate.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.fusesource.ide.server.karaf.core.server;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.fusesource.ide.server.karaf.core.server.subsystems.Karaf4xStartupLaunchConfigurator;
+import org.fusesource.ide.server.karaf.core.util.KarafUtils;
+import org.jboss.ide.eclipse.as.core.server.ILaunchConfigConfigurator;
+
+/**
+ * @author lheinema
+ *
+ */
+public class Karaf4xServerDelegate extends Karaf3xServerDelegate {
```

Comment:
wouldn't it be better to extract the common part of those classes in an abstract one and have specific bit of code for 3x and 4x in separated children class?

Reply:
I agree. Maybe we should consider a refactoring in future. (out of scope of this PR)

Reply:
I don't get why it is outside of the scope of this PR? the Karaf4xServerDelegate is introduced in this PR and the current structure is introducing technical debt, why not doing it right the first time?

Reply:
because it requires refactoring of the Karaf3xServerDelegate too. Lets create a refactor jira and tackle it separately.

Reply:
https://issues.jboss.org/browse/FUSETOOLS-2528